ignite/distributed/utils.py

```diff
@@ -522,7 +527,7 @@ def show_config():
     logger.info("node rank: {}".format(get_node_rank()))
 
 
-def one_rank_only(rank: int = 0, with_barrier: bool = False):
+def one_rank_only(rank: int = 0, with_barrier: bool = False) -> Optional[Callable]:
```

Comment:
Actually, again about Optional, why it is `Optional[Callable]` and not `Callable` ?

Reply:
Because of the `wrapper` function, which can be `None` if `get_rank() != rank`
https://github.com/pytorch/ignite/blob/b977b746edcbec9c29b8b88f40157f7d7db2601f/ignite/distributed/utils.py#L552-L564

Reply:
Here `ret` is the output of the function, I think wrapper is always defined but `func` is called only on specified rank.

Reply:
yeah, you are totally right. no idea, why I misread it 🙈